### ISSUE TITLE
DomainResourceRepository might ignore base

### ISSUE THREAD

**Issue by madaxel** [IchHabRecht/filefill#33]
https://github.com/IchHabRecht/filefill/blob/d746b0fa34243a2d00fba98bd30d2df597491c7f/Classes/Repository/DomainResourceRepository.php#L35

uses the getBase() method which returns the current Base for example depending on current context. If you are in DevelopmentContext the real domain would not be queried for the resource.

You might better use

`$url = $site->getConfiguration()['base'];`

**Comment by IchHabRecht** [comment]
Hi @madaxel,

Thank you for your issue. Would you mind to share the problem you have? Looking at the code for the site->base property the base is taken from the configuration. So there shouldn't be any different between the two.

**Comment by madaxel** [comment]
Hi @IchHabRecht ,
using local development context set to Development/Local with base url = test.local .
siteconfig.yml
```
base: 'https://www.production.org/'
baseVariants:
  -
    base: 'http://test.local/'
    condition: 'applicationContext == "Development/Local"'
  -
    base: 'https://test.org/'
    condition: 'applicationContext == "Production/Stage"'
  -
    base: 'https://integration.org/'
    condition: 'applicationContext == "Development"'
```
the system does not fetch files from https://www.production.org/ since in my case getBase()->toString() returns http://test.local/ and not as one might expect https://www.production.org/

**Comment by madaxel** [comment]
See typo3/sysext/core/Classes/Site/Entity/Site.php  Line 101ff

```        
 $baseUrl = $this->resolveBaseWithVariants(
            $configuration['base'] ?? '',
            $configuration['baseVariants'] ?? null
        );
        $this->base = new Uri($this->sanitizeBaseUrl($baseUrl));
```

**Comment by madaxel** [comment]
Hi @IchHabRecht ,
do you need more informations?

patch [filefill_33.txt](https://github.com/IchHabRecht/filefill/files/5694185/filefill_33.txt)

**Comment by IchHabRecht** [comment]
Hi @madaxel,

Thanks for sharing your information. All I needed was a little bit of my time to write the bugfix and release a new filefill version. The issue should be fixed with 3.3.1 now.
